interfaces/builtin/audio_playback.go

```diff
@@ -140,13 +142,24 @@ func (iface *audioPlaybackInterface) StaticInfo() interfaces.StaticInfo {
 	return interfaces.StaticInfo{
 		Summary:              audioPlaybackSummary,
 		ImplicitOnClassic:    true,
-		ImplicitOnCore:       true,
 		BaseDeclarationSlots: audioPlaybackBaseDeclarationSlots,
 	}
 }
 
 func (iface *audioPlaybackInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	spec.AddSnippet(audioPlaybackConnectedPlugAppArmor)
+
+	if slot.Snap().SnapName() != "snapd" {
+		// This enables audio in Core Desktop, by giving access to the
+		// sockets inside the snap's own folder.
+		instanceName := slot.Snap().InstanceName()
+		coreAppArmor := fmt.Sprintf(`
+owner /run/user/[0-9]*/snap.%s/pulse/ rw,
+owner /run/user/[0-9]*/snap.%s/pulse/native rwk,
+owner /run/user/[0-9]*/snap.%s/pulse/pid r,
+`, instanceName, instanceName, instanceName)
+		spec.AddSnippet(coreAppArmor)
+	}
 	if release.OnClassic {
 		spec.AddSnippet(audioPlaybackConnectedPlugAppArmorDesktop)
 	}
```

Comment:
/etc/pulse seems to be required, because, as you say, there are AppArmor denials. All the others aren't. I'll add that.

Reply:
The existing rules seem to be set up for "per-user PulseAudio on classic, system-wide PA on core", which kind of matched how the old pulseaudio snap worked but not how we're using the interfaces now.

I kind of wonder if it would be better to just merge the two snippets together and remove the OnClassic check.

Reply:
Mmm... these rules from `audioPlaybackConnectedPlugAppArmor`

    owner /{,var/}run/user/[0-9]*/pulse/ r,
    owner /{,var/}run/user/[0-9]*/pulse/native rwk,
    owner /{,var/}run/user/[0-9]*/pulse/pid r,

should allow to remove these ones from `audioPlaybackConnectedPlugAppArmorDesktop`:

    owner /{,var/}run/user/*/pulse/ rk,
    owner /{,var/}run/user/*/pulse/native rwk,
    owner /{,var/}run/user/*/pulse/pid r,

because the folders in /run/users/XXXX are, in our case, always [0-9]*.

About `/etc/pulse`, I think that it can be mixed for all plugs. I think that it's not a "security problem".

Finally, there is the question about the "pulse cookie". Is it still needed/relevant? The pipewire people says that it's not possible to configure that kind of authentication "at the moment", which sounds like it could be possible in a future... https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2399